tests/e2e-test.ts

```diff
@@ -53,7 +48,7 @@ export async function e2eTest({
     issuerConnectionId: senderRecipientConnection.id,
     credentialTemplate: {
       credentialDefinitionId: definition.id,
-      preview: previewFromAttributes({
+      preview: CredentialPreview.fromRecord({
```

Comment:
I like this improvement. I would consider calling it `fromJSON` to prevent confusion with other framework's records, that's what I would expect as a consumer at least. 

To me, the best API would be just passing JSON objects, but this is definitely step forward 👍

Reply:
The problem with `fromJSON` is that it clashes with `toJSON`.

Currently, `toJSON` transforms the class instance to the JSON representation. If we have a `fromJSON` method, I'd expect it to be the exact opposite of `toJSON`. Which it is not.

The reason I didn't change the constructor is because the `fromRecord` method removes the option to set a mime-type. Most time we just use text/plain, but it should be possible to set it

Reply:
I'm not set on `fromRecord`, so if you have suggestions I'm happy to create a new PR